it/test/controllers/UkimsNumberControllerSpec.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import helpers.ItTestBase
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.{WSClient, WSRequest}
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+
+class UkimsNumberControllerSpec extends ItTestBase {
+  lazy val client: WSClient = app.injector.instanceOf[WSClient]
+
+  private val url = s"http://localhost:$port$appRouteContext/ukims-number"
+
+  "Ukims number controller" should {
+
+    "redirects you to unauthorised page when auth fails" in {
+
+      noEnrolment
+
+      val request: WSRequest = client.url(url).withFollowRedirects(false)
+
+      val response = await(request.get())
+
+      response.status mustBe SEE_OTHER
+
+      redirectUrl(response) mustBe Some(routes.UnauthorisedController.onPageLoad.url)
+
+    }
+
+    "loads page" in {
+
+      authorisedUser
+
+      val request: WSRequest = client.url(url).withFollowRedirects(false)
+
+      val response = await(request.get())
+
+      response.status mustBe OK
+
+    }
+
+    "redirects to dummy controller when submitting valid data" in {
+
+      authorisedUser
+
+      val request: WSRequest = client.url(url).withFollowRedirects(false)
+
+      val response = await(request.post(Map("ukimsNumber" -> "XI47699357400020231115081800")))
+
+      response.status mustBe SEE_OTHER
+
+      redirectUrl(response) mustBe Some(routes.DummyController.onPageLoad.url)
+
+    }
+
+    "returns bad request when submitting no data" in {
+
+      authorisedUser
+
+      val request: WSRequest = client.url(url).withFollowRedirects(false)
+
+      val response = await(request.post(""))
+
+      response.status mustBe BAD_REQUEST
+
+    }
+  }
+}
```

Comment:
We need tests for scenario two:
Given I am on UK Internal Market Scheme number page
When I enter an incorrect format of my UKIMS number in the free text field
And I select continue
Then error message Enter your UKIMS number in the correct format should be displayed

Reply:
We also should try to follow the pattarn of controller tests seen in other branches

Reply:
Good point Ewan. New unit test added

Reply:
To verify that in the integration tests you can do something like:
  html(result)     should include("Enter your UKIMS number in the correct format should be displayed")

Reply:
This is added now